store/src/main/java/com/nytimes/android/external/store3/pipeline/PipelinePersister.kt

```diff
@@ -13,45 +13,44 @@ class PipelinePersister<Key, Input, Output>(
         private val writer: suspend (Key, Input) -> Unit,
         private val delete: (suspend (Key) -> Unit)? = null
 ) : PipelineStore<Key, Output> {
-    override suspend fun get(request: StoreRequest<Key>): Output? {
-        val value: Output? = if (request.shouldSkipCache(CacheType.DISK)) {
-            null
-        } else {
-            reader(request.key).singleOrNull()
-        }
-        value?.let {
-            return it
-        }
-        // skipped cache or cache is null
-        val fetcherValue = fetcher.get(request)
-        fetcherValue?.let {
-            writer(request.key, it)
-        }
-        return reader(request.key).singleOrNull()
-    }
-
     @Suppress("UNCHECKED_CAST")
     override fun stream(request: StoreRequest<Key>): Flow<Output> {
-        if (request.shouldSkipCache(CacheType.DISK)) {
-            return fetcher.stream(request)
-                .switchMap {
-                    writer(request.key, it)
-                    reader(request.key)
-                }.castNonNull()
-        } else {
-            return reader(request.key).let {
-                    if (request.refresh) {
-                        // also request from backend
-                        it.sideCollect(fetcher.stream(request)) { response: Input ->
-                            response?.let { data: Input ->
-                                writer(request.key, data)
-                            }
+        return when {
+            // skipping cache, just delegate to the fetcher but update disk w/ any new data from
+            // fetcher
+            request.shouldSkipCache(CacheType.DISK) -> fetcher.stream(request)
+                    .switchMap {
```

Comment:
```suggestion
                    .flatmap {
```

if new item comes from upstream we still want to cache the last one

Reply:
don't we want to stop the previous one and switch to the new one immediately ? because the `reader()` would never stop so i thought flatmap is not what we want. (i'll ping offline so that i can learn :) more RX)